src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -434,6 +440,13 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 	/* prevent concurrent placement changes */
 	AcquireMetadataLocks(taskList);
 
+	/*
+	 * We are taking locks on partitions of partitioned tables. These locks are
+	 * necessary for locking selected tables of router plannable INSERT/SELECT
```

Comment:
A suggestion, feel free to ignore:`These locks are necessary for locking tables that appear in the SELECT part of the query`